In [1]:
import geopandas as gp
import pandas as pd
import os
import math
import pber_functions_v1 as pdv
from functools import reduce

# Louisiana 2024 Primary Precinct-Level Election Results

## Load in Raw Files

In [2]:
# #Load in clean SOS file
# elections_2024_raw = pd.read_excel("./raw-from-source/Election+Results+(03-23-2024).xlsx",sheet_name="Multi-Parish(Precinct)")

# # Drop the first 4 rows
# elections_2024_raw = elections_2024_raw[4:]

# # Reset index
# elections_2024_raw.reset_index(inplace = True, drop = True)

# # Clean "Unnamed: 0" column
# elections_2024_raw["Unnamed: 0"] = elections_2024_raw["Unnamed: 0"].fillna("NONE")

# # Take a look at the file
# display(elections_2024_raw.head())


In [3]:
# slice_list = []
# prev = 0
# for val in range(1, len(elections_2024_raw)):
#     if elections_2024_raw.loc[val, "Unnamed: 0"]=="NONE" and elections_2024_raw.loc[val+1, "Unnamed: 0"]=="NONE" :
#         print(str(val))
#         print(str(val+1))
#         print(slice(val,val+1))
#         #slice_list.append(slice(val,val+1))
#         slice_list.append(slice(prev,val-1))
#         prev = val+2

In [4]:
# if not os.path.exists("./cleaned_results_primary/"):
#     os.mkdir("./cleaned_results_primary/")

In [5]:
# for slice_val in slice_list:
#     subset = elections_2024_raw.loc[slice_val]
#     subset.reset_index(inplace = True, drop = True)
#     contest = subset.loc[0,"Unnamed: 0"]
    
#     #if True:
#     print(contest)
#     if True:
#         #print(subset.loc[1])
#         subset.columns = [str(i)+ "-:-" + contest for i in subset.loc[1] ]
#         subset = subset.loc[2:]
#         subset.reset_index(inplace = True, drop = True)
#         #print(subset)
        
#         county = "Statewide"
#         county_dict = {}
#         for val in range(0, len(subset[subset.columns[1]])):
#             if math.isnan(subset.loc[val, subset.columns[1]]):
#                 #print(val)
#                 county = subset.loc[val,subset.columns[0]]
#             county_dict[val] = county
#         subset["COUNTY"] = subset.index.map(county_dict).fillna("BREAKPOINT")
#         assert(subset[subset["COUNTY"]=="BREAKPOINT"].shape[0]==0)
#         subset.rename(columns = {subset.columns[0]:"Precinct"}, inplace = True)        
#         subset = subset[~subset[subset.columns[1]].isna()]
        
#         subset.drop([i for i in subset.columns if i[:3] == "nan"], axis = 1, inplace = True)
        
#         first_column = subset.pop('COUNTY')
#         subset.insert(0, 'COUNTY', first_column)
        
#         # Allocate results
#         subset["UNIQUE_ID"] = subset["COUNTY"]+"-:-"+subset["Precinct"]
#         race_cols = [i for i in subset.columns if i not in ["UNIQUE_ID","COUNTY","Precinct"]]
#         #subset["PARISH"] = subset["UNIQUE_ID"].apply(lambda x: x.split("-:-")[0])
#         statewide_vals = subset[subset["COUNTY"]=="Statewide"]
#         print(statewide_vals.iloc[0])
#         subset = subset[subset["COUNTY"]!="Statewide"]
#         allocate_precs = subset[subset["UNIQUE_ID"].str.contains("Provisional") | subset["UNIQUE_ID"].str.contains("Early Voting")]
#         receiving_votes = subset[~(subset["UNIQUE_ID"].str.contains("Provisional") | subset["UNIQUE_ID"].str.contains("Early Voting"))]
        
#         subset_allocated = pdv.allocate_absentee(receiving_votes, allocate_precs, race_cols, "COUNTY", allocating_to_all_empty_precs=False)
        
        
#         subset_allocated.to_csv("./cleaned_results_primary/" + contest + ".csv", index = False)    

## Reload the Election Results File, Process So Each Row is a Precinct

In [6]:
path = "./raw-from-source/cleaned_results_primary/"
holder = []
for file in os.listdir(path):
    if file != ".DS_Store":
        temp = pd.read_csv(path+file)
        print(temp.columns)
        if file != "Presidential Nominee -- Democratic Party.csv":
            temp.drop(["COUNTY", "Precinct"], axis = 1, inplace = True)
        else:
            temp.rename(columns = {"COUNTY":"PARISH"}, inplace = True)
        holder.append(temp)

Index(['COUNTY', 'Precinct',
       'Marshall Meek (REP)-:-RSCC Member -- District 36C',
       'Chad Still (REP)-:-RSCC Member -- District 36C', 'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       '"Breka" Dennis (DEM)-:-DSCC Member --   4th Representative District, Office "A"',
       'Barbara M. Norton (DEM)-:-DSCC Member --   4th Representative District, Office "A"',
       'Lisa Tomkies (DEM)-:-DSCC Member --   4th Representative District, Office "A"',
       'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       'Kathleen "Kathy" Edgeworth (REP)-:-RSCC Member -- District 16A',
       'Christy Haik (REP)-:-RSCC Member -- District 16A', 'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       'Philip C. "Phil" Brickman (REP)-:-RSCC Member -- District  4B',
       'Adrian Bruneau (REP)-:-RSCC Member -- District  4B',
       'Juliet Laughlin (REP)-:-RSCC Member -- District  4B',
       'Nathaniel M. Phillips (REP)-:-RSCC Member -- D

Index(['COUNTY', 'Precinct',
       'Lyndon B. Johnson (DEM)-:-DSCC Member --   4th Representative District, Office "B"',
       'Ralph "RJ" Johnson (DEM)-:-DSCC Member --   4th Representative District, Office "B"',
       'Reginald Dewayne Johnson (DEM)-:-DSCC Member --   4th Representative District, Office "B"',
       'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       'Jeremy Aydell (REP)-:-RSCC Member -- District 18F',
       'Gregory Neff (REP)-:-RSCC Member -- District 18F', 'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       '"Nick" Carmena (REP)-:-RSCC Member -- District  6F',
       'Kimberly Powers (REP)-:-RSCC Member -- District  6F', 'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       'Christy Lynch (REP)-:-RSCC Member -- District  7B',
       'Donnie "Trey" Parker III (REP)-:-RSCC Member -- District  7B',
       'Jeffery Waltz (REP)-:-RSCC Member -- District  7B', 'UNIQUE_ID'],
      dtype='object')
Index(['COUNT

In [7]:
la_24_primary = reduce(lambda x, y: pd.merge(x, y, on = "UNIQUE_ID", how = "outer"), holder)
first_column = la_24_primary.pop('UNIQUE_ID')
la_24_primary.insert(0, 'UNIQUE_ID', first_column)
la_24_primary = la_24_primary.fillna(0)

/var/folders/1t/0q4w6hm92mg_zxd84dfxmq3m0000gn/T/ipykernel_3835/2051844483.py:3: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  la_24_primary.insert(0, 'UNIQUE_ID', first_column)


In [8]:
other_ones = [i for i in list(la_24_primary.columns) if i in ["UNIQUE_ID","PARISH","Precinct"] or "SCC" in i or "Presidential" in i]
la_24_primary = la_24_primary[other_ones]

In [9]:
la_24_primary_rename_dict = {i:i.replace("SCC Member --   ","SCC Member -- ") for i in list(la_24_primary.columns)}

In [10]:
la_24_primary.rename(columns = la_24_primary_rename_dict, inplace = True)

In [11]:
def get_race(race_string):
    race_string = race_string.title()
    race_string = race_string.replace("(Vote For 1)","")
    if "U.S. House" in race_string or 'Us House' in race_string or "Representative To Congress" in race_string or "U. S. Representative" in race_string:
        return "CON"
    elif "State House" in race_string or "State Representative" in race_string:
        return "SL"
    elif "State Senate" in race_string or "State Senator" in race_string:
        return "SU"
    elif "President" in race_string:
        return "PRE"
    elif "US Senate" in race_string or "Us Senate" in race_string or "U.S. Senator" in race_string or "U. S. Senator" in race_string:
        return "USS"
    elif "Public Service" in race_string:
        return "PSC"
    elif "Attorney General" in race_string:
        return "ATG"
    elif "Auditor General" in race_string or "Auditor" in race_string:
        return "AUD"
    elif "Treasurer" in race_string:
        return "TRE"
    elif "Superintendent" in race_string:
        return "SUP"
    elif "Secretary Of State" in race_string:
        return "SOS"
    elif "Lieutenant Governor" in race_string:
        return "LTG"
    elif "Governor" in race_string:
        return "GOV"
    elif "Commissioner Of Labor" in race_string:
        return "LAB"
    elif "Commissioner Of Agriculture" in race_string:
        return "AGR"
    elif "Commissioner Of Insurance" in race_string:
        return "INS"
    elif "Chief Justice Of The Supreme Court" in race_string:
        return "CJU"
    elif "Justice Of The Supreme Court" in race_string:
        return "JUS"
    elif "Ca No." in race_string:
        num = race_string.split(" ")[4][:-1]
        return "A"+str(num)
    elif "Rscc" in race_string or "Dscc" in race_string:
        return "SC"

    else:
        print("No race for:", race_string)
        raise ValueError
        
def get_election_type(race_string):
    if "CA No." in race_string:
        return ""
    else:
        return "R"
        
def get_party(race_string):
    if "(REP)" in race_string:
        return "R"
    elif "(DEM)" in race_string or "(Democrat)" in race_string:
        return "D"
    elif "(L)" in race_string or "(LBT)" in race_string:
        return "L"
    elif "(WI)" in race_string:
        return "O"
    elif "(IND)" in race_string:
        return "I"
    elif "(NOPTY)" in race_string:
        return "N"
    elif race_string[0:3]=="YES":
        return "YES"
    elif race_string[0:2]=="NO":
        return "NO"
    else:
        return "O"
           
def get_name(name_string):
    if "No." in name_string:
        return ""
    elif " (" in name_string and "Amendment" not in name_string and "Referendum" not in name_string:
        #print(name_string)
        name_string = name_string.split(" (")[0]
        name_string = name_string.replace("'","")
        likely_last = name_string.split(" ")[-1]
        proposed_last = likely_last[:3]
        if proposed_last in ['II', 'III', 'Jr', 'Jr.', 'Sr.', 'JR.', "JR", "IV", 'I']:
            likely_last = name_string.split(" ")[-2]
            proposed_last = likely_last[:3]
        #print(proposed_last.upper())
        return proposed_last.upper()
    else:
        return name_string.split(" ")[-1][:3].upper()
import re
 

def get_district(race_string, fill_level):
    temp = race_string.split(" -- ")[1].strip().split(" ")[0]
    return re.sub("\D","",temp).zfill(fill_level)

def get_district_advanced(race_string):
    if " --  " in race_string:
        split = " --  "
    else:
        split = " -- "
    if "Representative" in race_string:
        half = race_string.split(split)[1]
        num = half.split(" ")[0][:-2]
        char = half.split(" ")[-1][1]
        return (num+char).zfill(4)
    else:
        return race_string.split(" ")[-1].zfill(4)
    

def column_rename_function(name_string):
    election_type = "P"
    year = "24"
    party = get_party(name_string)
    race = get_race(name_string)
    district = ""
    #print(name_string)
    if race in ["CON", "SU"]:
        district = get_district(name_string, 2)
        year = ""
    elif race in ["SL"]:
        district = get_district(name_string, 3)
        year = ""
    elif race in ["SC"]:
        year = ""
        district = get_district_advanced(name_string)
        election_type = ""
    name = get_name(name_string)
    new_col_name = election_type + year + race + district + party + name
#     print("election_type: ", election_type)
#     print("year: ", year)
#     print("race: ", race)
#     print("district: ", district)
#     print("party: ", party)
#     print("name: ", name)
    if len(new_col_name) > 10:
        print("LONG NAME")
        print(name_string, "->", new_col_name)
    return new_col_name

races_to_clean = [i for i in la_24_primary.columns if i not in ["UNIQUE_ID", "PARISH", "Precinct"]]


race_updates_dict = {}
race_updates_reversed = {}
clean_dups = {}
new_names = []
for val in races_to_clean:
    new_name = column_rename_function(val)
    race_updates_dict[val] = new_name
    if new_name not in new_names:
        new_names.append(new_name)
        race_updates_reversed[new_name] = val
    else:
        print("Duplicate", new_name)
        print("\t",race_updates_reversed[new_name])
        print("\t",val)
        print("")
        clean_dups[val] = race_updates_reversed[new_name]

Duplicate SC016CRGRA
	 Conner Graham (REP)-:-RSCC Member -- District 16C
	 Lynn Coxe Graham (REP)-:-RSCC Member -- District 16C

Duplicate SC004BDJOH
	 Lyndon B. Johnson (DEM)-:-DSCC Member -- 4th Representative District, Office "B"
	 Ralph "RJ" Johnson (DEM)-:-DSCC Member -- 4th Representative District, Office "B"

Duplicate SC004BDJOH
	 Lyndon B. Johnson (DEM)-:-DSCC Member -- 4th Representative District, Office "B"
	 Reginald Dewayne Johnson (DEM)-:-DSCC Member -- 4th Representative District, Office "B"



In [12]:
race_updates_dict["Conner Graham (REP)-:-RSCC Member -- District 16C"] = "SC016CRCGR"
race_updates_dict['''Ralph "RJ" Johnson (DEM)-:-DSCC Member -- 4th Representative District, Office "B"'''] = "SC004BDRJO"
race_updates_dict['''Reginald Dewayne Johnson (DEM)-:-DSCC Member -- 4th Representative District, Office "B"'''] = "SC004BDDJO"
race_updates_dict['''Omari J. Ho-Sang (DEM)-:-DSCC Member -- 9th Representative District, Office "A"'''] = 'SC009ADHOS'

In [13]:
race_updates_dict

{'Marshall Meek (REP)-:-RSCC Member -- District 36C': 'SC036CRMEE',
 'Chad Still (REP)-:-RSCC Member -- District 36C': 'SC036CRSTI',
 '"Breka" Dennis (DEM)-:-DSCC Member -- 4th Representative District, Office "A"': 'SC004ADDEN',
 'Barbara M. Norton (DEM)-:-DSCC Member -- 4th Representative District, Office "A"': 'SC004ADNOR',
 'Lisa Tomkies (DEM)-:-DSCC Member -- 4th Representative District, Office "A"': 'SC004ADTOM',
 'Kathleen "Kathy" Edgeworth (REP)-:-RSCC Member -- District 16A': 'SC016AREDG',
 'Christy Haik (REP)-:-RSCC Member -- District 16A': 'SC016ARHAI',
 'Philip C. "Phil" Brickman (REP)-:-RSCC Member -- District  4B': 'SC004BRBRI',
 'Adrian Bruneau (REP)-:-RSCC Member -- District  4B': 'SC004BRBRU',
 'Juliet Laughlin (REP)-:-RSCC Member -- District  4B': 'SC004BRLAU',
 'Nathaniel M. Phillips (REP)-:-RSCC Member -- District  4B': 'SC004BRPHI',
 'Anne Queyrouze (REP)-:-RSCC Member -- District  4B': 'SC004BRQUE',
 'Anthony L. Christy (DEM)-:-DSCC Member --  59th Representative D

In [14]:
# Export the Renamed Columns for the README
pd.DataFrame(race_updates_dict.items()).to_csv("./field_names_prim.csv", index = False)

# Rename the columns in the election results file
la_24_primary.rename(columns = race_updates_dict, inplace = True)

In [15]:
la_24_primary["SC004BDRJO"]

0       0.0
1       0.0
2       0.0
3       0.0
4       0.0
       ... 
3828    0.0
3829    0.0
3830    0.0
3831    0.0
3832    0.0
Name: SC004BDRJO, Length: 3833, dtype: float64

In [16]:
la_24_comb = la_24_primary.copy(deep = True)

In [17]:
fips_file = pd.read_csv("./raw-from-source/FIPS/US_FIPS_Codes.csv")
fips_file = fips_file[fips_file["State"]=="Louisiana"]
fips_file["FIPS County"] = fips_file["FIPS County"].astype(str)
fips_file["FIPS County"] = fips_file["FIPS County"].str.zfill(3)
fips_file["County Name"] = fips_file["County Name"].replace("La Salle","Lasalle")
fips_dict = dict(zip(fips_file['County Name'], fips_file['FIPS County']))
la_24_comb['COUNTYFP'] = la_24_comb['PARISH'].str.replace("St.","St").map(fips_dict).fillna(la_24_comb['PARISH'])
la_24_comb['COUNTYFP'] = la_24_comb['COUNTYFP'].astype(str).str.zfill(3)

# Print statements to check the county FIPs we've added
print(la_24_comb['COUNTYFP'].unique())

['119' '017' '033' '071' '005' '109' '057' '077' '097' '007' '047' '103'
 '079' '105' '055' '099' '073' '051' '101' '015' '081' '063' '019' '001'
 '003' '009' '011' '013' '021' '023' '025' '027' '029' '031' '035' '037'
 '039' '041' '043' '045' '049' '053' '059' '061' '065' '067' '069' '075'
 '083' '085' '087' '089' '091' '093' '095' '107' '111' '113' '115' '117'
 '121' '123' '125' '127']


/var/folders/1t/0q4w6hm92mg_zxd84dfxmq3m0000gn/T/ipykernel_3835/1836122395.py:7: FutureWarning: The default value of regex will change from True to False in a future version.
  la_24_comb['COUNTYFP'] = la_24_comb['PARISH'].str.replace("St.","St").map(fips_dict).fillna(la_24_comb['PARISH'])


In [18]:
# Sort the race columns in alphabetical order
race_cols = list(race_updates_dict.values())
race_cols.sort()

In [19]:
# Print out the vote totals of all of the races for checking
for race in race_cols:
    #print(race)
    la_24_comb.loc[:,race] = la_24_comb.loc[:,race].astype(int)
    print("~~"+race, sum(la_24_comb[race]))

/var/folders/1t/0q4w6hm92mg_zxd84dfxmq3m0000gn/T/ipykernel_3835/942614151.py:4: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  la_24_comb.loc[:,race] = la_24_comb.loc[:,race].astype(int)


~~P24PREDBID 143380
~~P24PREDELY 2652
~~P24PREDLOZ 2245
~~P24PREDLYO 3770
~~P24PREDPER 1200
~~P24PREDPHI 4351
~~P24PREDUYG 1114
~~P24PREDWIL 7898
~~P24PRERBIN 580
~~P24PRERCHR 1281
~~P24PRERDES 3022
~~P24PRERHAL 13123
~~P24PRERHUT 519
~~P24PRERRAM 595
~~P24PRERSTU 210
~~P24PRERSWI 335
~~P24PRERTRU 172503
~~SC001ADBOW 1856
~~SC001ADSMI 1513
~~SC001ARALP 564
~~SC001ARLIO 305
~~SC001BDBOW 1912
~~SC001BDLES 1397
~~SC001CRDEE 439
~~SC001CRSMI 539
~~SC001ERDUB 449
~~SC001ERHIR 310
~~SC001ERSIG 188
~~SC001GRBAY 378
~~SC001GRGAR 184
~~SC001GROWE 206
~~SC002ADGIL 2722
~~SC002ADHAR 2125
~~SC002ARBER 182
~~SC002ARCAL 259
~~SC002ARPRU 263
~~SC002BRKAT 266
~~SC002BRLEE 314
~~SC003ARENS 58
~~SC003ARHAR 61
~~SC003ARPHI 21
~~SC003ARSTE 123
~~SC003ARSWA 57
~~SC003BRLUN 325
~~SC003BROST 144
~~SC004ADDEN 1411
~~SC004ADNOR 4748
~~SC004ADTOM 1212
~~SC004BDDJO 2071
~~SC004BDJOH 4187
~~SC004BDRJO 728
~~SC004BRBRI 144
~~SC004BRBRU 138
~~SC004BRLAU 51
~~SC004BRPHI 28
~~SC004BRQUE 67
~~SC005ADCAL 972
~~SC005ADP

~~P24PREDBID 143380~~
~~P24PREDELY 2652~~
~~P24PREDLOZ 2245~~
~~P24PREDLYO 3770~~
~~P24PREDPER 1200~~
~~P24PREDPHI 4351~~
~~P24PREDUYG 1114~~
~~P24PREDWIL 7898~~
~~P24PRERBIN 580~~
~~P24PRERCHR 1281~~
~~P24PRERDES 3022~~
~~P24PRERHAL 13123~~
~~P24PRERHUT 519~~
~~P24PRERRAM 595~~
~~P24PRERSTU 210~~
~~P24PRERSWI 335~~
~~P24PRERTRU 172503~~

~~SC001ADBOW 1856~~
~~SC001ADSMI 1513~~
~~SC001ARALP 564~~
~~SC001ARLIO 305~~
~~SC001BDBOW 1912~~
~~SC001BDLES 1397~~
~~SC001CRDEE 439~~
~~SC001CRSMI 539~~
~~SC001ERDUB 449~~
~~SC001ERHIR 310~~
~~SC001ERSIG 188~~
~~SC001GRBAY 378~~
~~SC001GRGAR 184~~
~~SC001GROWE 206~~
~~SC002ADGIL 2722~~
~~SC002ADHAR 2125~~
~~SC002ARBER 182~~
~~SC002ARCAL 259~~
~~SC002ARPRU 263~~
~~SC002BRKAT 266~~
~~SC002BRLEE 314~~
~~SC003ARENS 58~~
~~SC003ARHAR 61~~
~~SC003ARPHI 21~~
~~SC003ARSTE 123~~
~~SC003ARSWA 57~~
~~SC003BRLUN 325~~
~~SC003BROST 144~~
~~SC004ADDEN 1411~~
~~SC004ADNOR 4748~~
~~SC004ADTOM 1212~~
~~SC004BDDJO 2071~~
~~SC004BDJOH 4187~~
~~SC004BDRJO 728~~
~~SC004BRBRI 144~~
~~SC004BRBRU 138~~
~~SC004BRLAU 51~~
~~SC004BRPHI 28~~
~~SC004BRQUE 67~~
~~SC005ADCAL 972~~
~~SC005ADPOS 549~~
~~SC005ARFOS 43~~
~~SC005ARKEP 123~~
~~SC005ARRUO 97~~
~~SC005BDCOX 939~~
~~SC005BDJEA 564~~
~~SC005BRCOO 150~~
~~SC005BRMCE 123~~
~~SC006ADMER 588~~
~~SC006ADOLD 954~~
~~SC006ADSTE 1003~~
~~SC006BDCOT 1203~~
~~SC006BDDAR 1427~~
~~SC006BRLIN 311~~
~~SC006BRSTA 228~~
~~SC006CRKRO 242~~
~~SC006CRWIL 423~~
~~SC006DRCRA 138~~
~~SC006DRMES 543~~
~~SC006ERIVE 509~~
~~SC006ERLAZ 229~~
~~SC006FRCAR 372~~
~~SC006FRPOW 307~~
~~SC006HRBIN 234~~
~~SC006HRMAC 472~~
~~SC007BRLYN 249~~
~~SC007BRPAR 148~~
~~SC007BRWAL 115~~
~~SC008BDLEI 328~~
~~SC008BDSER 261~~
~~SC008CRCAN 402~~
~~SC008CRPRO 184~~
~~SC008CRYOR 125~~
~~SC008DRCLA 317~~
~~SC008DRKIR 370~~
~~SC009ADBAK 553~~
~~SC009ADHOS 84~~
~~SC009ARLEO 478~~
~~SC009ARSER 468~~
~~SC009BRBRO 373~~
~~SC009BRDON 261~~
~~SC009BRTHE 285~~
~~SC009FRBER 436~~
~~SC009FRKIR 721~~
~~SC010BRCAL 117~~
~~SC010BRLEB 814~~
~~SC010BRZIM 152~~
~~SC010DRCAP 359~~
~~SC010DRGUI 449~~
~~SC010FRBAU 522~~
~~SC010FRSOU 288~~
~~SC010GRBRA 712~~
~~SC010GRODE 254~~
~~SC011ADMAC 178~~
~~SC011ADWIL 1255~~
~~SC011ARHAM 502~~
~~SC011ARSTR 523~~
~~SC011BDAND 484~~
~~SC011BDWOO 114~~
~~SC011BDYOU 855~~
~~SC011CRFER 485~~
~~SC011CRHOT 561~~
~~SC011FRRAY 598~~
~~SC011FRSCU 431~~
~~SC011GRGAL 817~~
~~SC011GRSHI 156~~
~~SC011KRBON 339~~
~~SC011KRCAS 525~~
~~SC011KRTAT 492~~
~~SC012ARCHR 272~~
~~SC012ARVAR 285~~
~~SC012BDCAB 194~~
~~SC012BDEUN 169~~
~~SC012BDSKA 397~~
~~SC012ERBRE 570~~
~~SC012ERFRI 684~~
~~SC013ARKEE 359~~
~~SC013ARLAS 348~~
~~SC013DRBAB 782~~
~~SC013DRCRO 292~~
~~SC013GRAND 218~~
~~SC013GRMOR 469~~
~~SC014ARCHI 178~~
~~SC014ARJER 43~~
~~SC014ARKIT 94~~
~~SC014ARSAV 113~~
~~SC014BDALL 647~~
~~SC014BDNOE 895~~
~~SC016AREDG 222~~
~~SC016ARHAI 277~~
~~SC016CRCGR 169~~
~~SC016CRGRA 253~~
~~SC016CRMCC 95~~
~~SC017ARFAB 408~~
~~SC017ARLEM 266~~
~~SC017DRHUV 381~~
~~SC017DRPEY 110~~
~~SC017FRBIL 134~~
~~SC017FRHAR 320~~
~~SC018ADBEN 764~~
~~SC018ADPAU 594~~
~~SC018ARBEN 225~~
~~SC018ARFOR 259~~
~~SC018CRALO 211~~
~~SC018CRCAR 71~~
~~SC018CRFOS 147~~
~~SC018DRLAM 385~~
~~SC018DRSIM 228~~
~~SC018FRAYD 625~~
~~SC018FRNEF 136~~
~~SC019ARDUE 432~~
~~SC019ARMON 326~~
~~SC019ERJAM 241~~
~~SC019ERSUM 337~~
~~SC020BRBER 244~~
~~SC020BRWHI 424~~
~~SC020DRLEC 450~~
~~SC020DRTHI 158~~
~~SC020ERTRO 418~~
~~SC020ERWAL 184~~
~~SC020FRBAZ 459~~
~~SC020FRENG 152~~
~~SC021ARGUL 379~~
~~SC021ARHOW 151~~
~~SC021BDCON 302~~
~~SC021BDDAV 699~~
~~SC021BDJOH 1496~~
~~SC021BDSIN 51~~
~~SC021BDWIL 161~~
~~SC021FRCAF 410~~
~~SC021FRPUR 108~~
~~SC022ARBRE 251~~
~~SC022ARROM 486~~
~~SC023ADCLE 412~~
~~SC023ADCON 1130~~
~~SC023ADGRE 1089~~
~~SC023ARRIC 442~~
~~SC023ARSKI 208~~
~~SC023BDGER 266~~
~~SC023BDMEN 1180~~
~~SC023BDSWE 1111~~
~~SC023BRANG 263~~
~~SC023BRBEN 207~~
~~SC023BRCAM 95~~
~~SC023BRESC 179~~
~~SC023CRLIN 281~~
~~SC023CRSUP 271~~
~~SC023DRCAM 353~~
~~SC023DRNOR 191~~
~~SC023FREAT 264~~
~~SC023FRIST 138~~
~~SC023JRBUC 440~~
~~SC023JRPER 176~~
~~SC024DRKIS 352~~
~~SC024DRREE 147~~
~~SC025ADSAN 1129~~
~~SC025ADSUT 303~~
~~SC026ADWAR 1239~~
~~SC026ADWHI 492~~
~~SC026BRHAR 71~~
~~SC026BRNAQ 310~~
~~SC026CRHOO 73~~
~~SC026CRSAV 307~~
~~SC026HRCRA 459~~
~~SC026HRTOT 127~~
~~SC028ADMAR 930~~
~~SC028ADOBR 180~~
~~SC028BDMCK 401~~
~~SC028BDWIL 709~~
~~SC028FRCAV 479~~
~~SC028FRJOH 483~~
~~SC029BDDOG 231~~
~~SC029BDJOR 1241~~
~~SC029BDLAN 925~~
~~SC029BDRIC 259~~
~~SC031CREDW 456~~
~~SC031CRPRI 308~~
~~SC031GRDAV 477~~
~~SC031GRWRI 592~~
~~SC032ARHAR 507~~
~~SC032ARMON 328~~
~~SC032CRDEL 326~~
~~SC032CRPHI 564~~
~~SC032GRAVE 508~~
~~SC032GRPAN 155~~
~~SC032HRHAR 302~~
~~SC032HRTAR 549~~
~~SC033FRDAV 378~~
~~SC033FRSTA 117~~
~~SC034BDFON 601~~
~~SC034BDGUI 203~~
~~SC034BDSMI 765~~
~~SC035ARBRO 440~~
~~SC035ARHIN 224~~
~~SC035FRNUN 486~~
~~SC035FRWHE 206~~
~~SC035HRBRO 305~~
~~SC035HRCOO 114~~
~~SC035HRTAR 417~~
~~SC036CRMEE 222~~
~~SC036CRSTI 389~~
~~SC036DRLOM 252~~
~~SC036DRWYC 254~~
~~SC037DRBAI 326~~
~~SC037DRBAR 270~~
~~SC037FRMCI 529~~
~~SC037FRSMI 655~~
~~SC038FRAVA 1476~~
~~SC038FRROG 635~~
~~SC039CRERS 570~~
~~SC039CRSAV 642~~
~~SC040BDCLA 866~~
~~SC040BDRIC 922~~
~~SC044ADBRO 1096~~
~~SC044ADMIC 336~~
~~SC044BDDAL 414~~
~~SC044BDEDM 998~~
~~SC045ADBER 338~~
~~SC045ADCLA 601~~
~~SC046ADDES 528~~
~~SC046ADGAL 156~~
~~SC050BDCON 494~~
~~SC050BDPAR 684~~
~~SC051BDCAS 335~~
~~SC051BDTRI 227~~
~~SC056BDDAV 398~~
~~SC056BDPRI 460~~
~~SC057BDCOO 552~~
~~SC057BDGAI 911~~
~~SC057BDSOR 452~~
~~SC058ADBAN 570~~
~~SC058ADDUM 1261~~
~~SC058ADFRA 345~~
~~SC058ADROU 281~~
~~SC059BDCHR 528~~
~~SC059BDHER 245~~
~~SC060BDGRA 1272~~
~~SC060BDMOR 951~~
~~SC061ADCON 640~~
~~SC061ADMAR 1831~~
~~SC061BDBEL 916~~
~~SC061BDSMI 1147~~
~~SC061BDYAN 298~~
~~SC062BDBAN 1261~~
~~SC062BDKEN 798~~
~~SC062BDLOV 603~~
~~SC063ADMIL 2278~~
~~SC063ADWRI 1093~~
~~SC063BDBLA 1044~~
~~SC063BDJOH 2170~~
~~SC065ADADK 627~~
~~SC065ADWAM 224~~
~~SC066ADLEG 332~~
~~SC066ADMAT 198~~
~~SC066ADPRI 512~~
~~SC066BDHOL 602~~
~~SC066BDTAY 408~~
~~SC067ADCUL 1005~~
~~SC067ADHOL 564~~
~~SC067BDCAS 709~~
~~SC067BDLEW 824~~
~~SC068ADDAV 1056~~
~~SC068ADHAR 353~~
~~SC068ADRUS 160~~
~~SC068BDADA 846~~
~~SC068BDSCH 626~~
~~SC069ADBOA 417~~
~~SC069ADELL 466~~
~~SC069ADVIC 287~~
~~SC069BDFRI 437~~
~~SC069BDJEF 693~~
~~SC071BDGON 192~~
~~SC071BDRIL 224~~
~~SC076BDMIL 789~~
~~SC076BDPER 522~~
~~SC077BDLEW 569~~
~~SC077BDTHO 633~~
~~SC080BDBAI 388~~
~~SC080BDMCG 531~~
~~SC081ADHAR 401~~
~~SC081ADHYD 315~~
~~SC083ADMAN 661~~
~~SC083ADWAT 774~~
~~SC083BDDIC 304~~
~~SC083BDGRE 947~~
~~SC083BDJON 257~~
~~SC084ADMIT 424~~
~~SC084ADSMI 516~~
~~SC084BDFLO 177~~
~~SC084BDLEE 620~~
~~SC084BDMIT 187~~
~~SC085ADEWE 405~~
~~SC085ADMAN 475~~
~~SC085BDBLA 355~~
~~SC085BDSMI 546~~
~~SC086BDFLE 258~~
~~SC086BDMUS 308~~
~~SC087ADLYO 939~~
~~SC087ADMCG 722~~
~~SC087BDCAL 229~~
~~SC087BDLYO 757~~
~~SC087BDSHE 757~~
~~SC089BDFON 532~~
~~SC089BDMIL 521~~
~~SC090ADBUR 337~~
~~SC090ADDUR 271~~
~~SC090ADGRE 588~~
~~SC091ADAND 1236~~
~~SC091ADBAJ 1419~~
~~SC091BDCAR 626~~
~~SC091BDFOR 1212~~
~~SC091BDPHI 699~~
~~SC092ADDUN 626~~
~~SC092ADSHA 539~~
~~SC092BDHUT 555~~
~~SC092BDPRI 501~~
~~SC093BDDAV 829~~
~~SC093BDDUP 2071~~
~~SC094ADLUL 487~~
~~SC094ADSAV 711~~
~~SC096ADPIE 1094~~
~~SC096ADVEN 374~~
~~SC096BDLEW 1177~~
~~SC096BDXAN 232~~
~~SC097ADTHO 3280~~
~~SC097ADWIL 1011~~
~~SC097BDBAR 945~~
~~SC097BDGRE 2536~~
~~SC097BDMOR 1092~~
~~SC098ADCHE 558~~
~~SC098ADFAY 739~~
~~SC098ADRAT 365~~
~~SC098ADSTE 1074~~
~~SC098BDSIE 825~~
~~SC098BDZER 1794~~
~~SC099ADGLE 1021~~
~~SC099ADNEW 2492~~
~~SC099BDEXP 1927~~
~~SC099BDHEN 1527~~
~~SC102ADALL 603~~
~~SC102ADBOY 2530~~
~~SC102BDCAR 2686~~
~~SC102BDMOS 455~~
~~SC103BDDUV 186~~
~~SC103BDRIL 462~~

In [20]:
la_24_comb.rename(columns = {"PARISH":"Parish"}, inplace = True)
la_24_comb = la_24_comb[["UNIQUE_ID", "COUNTYFP", 'Parish', "Precinct"]+ race_cols]

In [28]:
la_24_comb["COUNTYFP"].unique()

array(['119', '017', '033', '071', '005', '109', '057', '077', '097',
       '007', '047', '103', '079', '105', '055', '099', '073', '051',
       '101', '015', '081', '063', '019', '001', '003', '009', '011',
       '013', '021', '023', '025', '027', '029', '031', '035', '037',
       '039', '041', '043', '045', '049', '053', '059', '061', '065',
       '067', '069', '075', '083', '085', '087', '089', '091', '093',
       '095', '107', '111', '113', '115', '117', '121', '123', '125',
       '127'], dtype=object)

In [22]:
la_24_comb.to_csv("./la_2024_prim_prec/la_2024_prim_prec.csv", index = False)

In [23]:
import shutil

In [26]:
shutil.make_archive("./la_2024_prim_prec","zip","./la_2024_prim_prec")

'/Users/peterhorton/Documents/RDH/pber_local/LA_2024/la_2024_prim_prec.zip'